diff -urNad linux-2.6.34-rc5/arch/x86/kernel/relocate_kernel_32.S linux-2.6.34-rc5-bb/arch/x86/kernel/relocate_kernel_32.S
--- linux-2.6.34-rc5/arch/x86/kernel/relocate_kernel_32.S	2010-04-19 20:29:56.000000000 -0300
+++ linux-2.6.34-rc5-bb/arch/x86/kernel/relocate_kernel_32.S	2010-04-24 17:44:36.706329661 -0300
@@ -54,8 +54,10 @@
 	movl	%eax, CR0(%edi)
 	movl	%cr3, %eax
 	movl	%eax, CR3(%edi)
+#ifndef CONFIG_X86_RDC321X
 	movl	%cr4, %eax
 	movl	%eax, CR4(%edi)
+#endif
 
 	/* read the arguments and say goodbye to the stack */
 	movl  20+4(%esp), %ebx /* page_list */
@@ -122,7 +124,9 @@
 	 * Setting everything to zero seems safe.
 	 */
 	xorl	%eax, %eax
+#ifndef CONFIG_X86_RDC321X
 	movl	%eax, %cr4
+#endif
 
 	jmp 1f
 1:
@@ -193,8 +197,10 @@
 	ret
 
 virtual_mapped:
+#ifndef CONFIG_X86_RDC321X
 	movl	CR4(%edi), %eax
 	movl	%eax, %cr4
+#endif
 	movl	CR3(%edi), %eax
 	movl	%eax, %cr3
 	movl	CR0(%edi), %eax
